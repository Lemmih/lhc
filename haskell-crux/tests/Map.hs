module Map where

map :: (a -> b) -> [a] -> [b]
map f lst =
  case lst of
    []     -> []
    (x:xs) -> f x : map f xs
