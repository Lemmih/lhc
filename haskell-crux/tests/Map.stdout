Map.map : ∀ sk_0_a sk_1_b. (sk_0_a →  sk_1_b) →  [sk_0_a] →  [sk_1_b]
Map.map =
  λ @.sk_0_a^1:* @.sk_1_b^2:* Map.f^3:(sk_0_a →  sk_1_b) Map.lst^4:[sk_0_a] → 
  case Map.lst^4 of
    scrut^5:[sk_0_a]
    LHC.Prim.Nil  → 
      LHC.Prim.Nil
    LHC.Prim.Cons Map.x^6:sk_0_a Map.xs^7:[sk_0_a] → 
      LHC.Prim.Cons (Map.f^3
            Map.x^6) (Map.map @.sk_0_a^1
              @.sk_1_b^2 Map.f^3 Map.xs^7)
